src/dawn/Optimizer/PassSetNonTempCaches.h

```diff
@@ -0,0 +1,47 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                          _
+//                         | |
+//                       __| | __ ___      ___ ___
+//                      / _` |/ _` \ \ /\ / / '_  |
+//                     | (_| | (_| |\ V  V /| | | |
+//                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+
+#ifndef DAWN_OPTIMIZER_PASSNONTEMPCACHES_H
+#define DAWN_OPTIMIZER_PASSNONTEMPCACHES_H
+
+#include "dawn/Optimizer/Pass.h"
+
+namespace dawn {
+
+/// @brief This Pass optimizes data locality on a per-multistage basis
+///
+/// For all the non-temporary fields that we access in any given multistage, we check how high the
```

Comment:
For all the non-temporary, vertically-pointwise fields ... Then you can skip the next sentence